Sprint 17

GrachevAl opened the pull request.

GrachevAl committed:
Добавлены модули зависимости по каждому экрану

GrachevAl committed:
Добавлены модули зависимости по каждому экрану версия 1

GrachevAl committed:
Добавлены модули зависимости по каждому экрану версия 2

vlad-zotov commented on app/src/main/java/com/example/playlistmaker/di/SearchModule.kt:
```diff
@@ -0,0 +1,60 @@
+            .getSharedPreferences(Const.SHARED_PREFERENCES_HISTORY_LIST, Context.MODE_PRIVATE)
+    }
+
+    single<TrackStorage> { TrackStorageImpl(sharedPreferences = get(qualifier = named(Const.SHARED_PREFERENCES_HISTORY_LIST) )) }
```
Привет! Для удобства чтения кода, такие конструкции лучше разносить на несколько строк

vlad-zotov commented on app/src/main/java/com/example/playlistmaker/di/SearchModule.kt:
```diff
@@ -0,0 +1,60 @@
+    }
+
+    single<TrackStorage> { TrackStorageImpl(sharedPreferences = get(qualifier = named(Const.SHARED_PREFERENCES_HISTORY_LIST) )) }
+    single<TrackApi>(named("api")){
```
хорошим подходом будет вынести `"api"` в константу и переиспользовать в проекте, чтобы не дублировать код

vlad-zotov commented on app/src/main/java/com/example/playlistmaker/di/SearchModule.kt:
```diff
@@ -0,0 +1,60 @@
+        )
+    }
+}
+
```
пустые строчки в конце файла лишние

vlad-zotov commented on app/src/main/java/com/example/playlistmaker/di/ShareModule.kt:
```diff
@@ -0,0 +1,29 @@
+    factory { OpenTerms(sharingRepository = get()) }
+    factory { SendToSupport(sharingRepository = get()) }
+    factory { ShareApp(sharingRepository = get()) }
+
```
пустые строки могут улучшить читаемость кода, но не в данном случае. Стоит их удалить отсюда.

vlad-zotov commented on app/src/main/java/com/example/playlistmaker/player/ui/AudioPlayerActivity.kt:
```diff
@@ -28,7 +29,8 @@ class AudioPlayerActivity() : AppCompatActivity() {
     }
 
-    private lateinit var viewModel: AudioPlayerViewModel
+    //private lateinit var viewModel: AudioPlayerViewModel
```
старый закомментированный код следует удалить

vlad-zotov commented on app/src/main/java/com/example/playlistmaker/player/ui/AudioPlayerActivity.kt:
```diff
@@ -46,7 +48,7 @@ class AudioPlayerActivity() : AppCompatActivity() {
         }
 
-        viewModel = ViewModelProvider(this, AudioPlayerViewModelFactory())[AudioPlayerViewModel::class.java]
+        //viewModel = ViewModelProvider(this, AudioPlayerViewModelFactory())[AudioPlayerViewModel::class.java]
```
старый закомментированный код следует удалить

vlad-zotov commented on app/src/main/java/com/example/playlistmaker/search/ui/SearchScreen.kt:
```diff
@@ -19,20 +19,31 @@ import com.example.playlistmaker.*
     val adapterHistoryList = TrackAdapter(this)
 
-    private lateinit var searchViewModel: SearchScreenViewModel
+    //private lateinit var searchViewModel: SearchScreenViewModel
```
старый закомментированный код следует удалить

vlad-zotov commented on app/src/main/java/com/example/playlistmaker/search/data/storage/TrackStorageImpl.kt:
```diff
@@ -11,6 +11,8 @@ import com.google.gson.reflect.TypeToken
 
 class TrackStorageImpl(private val sharedPreferences: SharedPreferences) : TrackStorage {
 
+
```
здесь пустые строчки тоже не несут полезной нагрузки. Лучше почистить

vlad-zotov commented on app/src/main/java/com/example/playlistmaker/settings/ui/SettingsActivity.kt:
```diff
@@ -19,16 +20,15 @@ class SettingsActivity : AppCompatActivity() {
-    private lateinit var viewModelSetting: SettingViewModel
-
-
+    //private lateinit var viewModelSetting: SettingViewModel
```
не бойся удалять старый код, если новый хорошо работает. В случае необходимости всегда можно откатить код благодаря Git

GrachevAl committed:
Спринт 17:
1. Добавлены модули зависимости по каждому экрану версия 2
2. Исправлены баги с отрисовкой View элементов

GrachevAl committed:
Спринт 17 ver 1:
1. Добавлены модули зависимости по каждому экрану версия 2
2. Исправлены баги с отрисовкой View элементов
